crates/precompiles/src/tip_fee_manager/amm.rs

```diff
@@ -54,6 +48,16 @@ pub struct PoolKey {
     pub validator_token: Address,
 }
 
+// TODO(rusowsky): remove this and create a read-only wrapper that is callable from read-only ctx with db access
+impl Pool {
+    pub fn decode_from_slot(slot_value: U256) -> Self {
+        use crate::storage::{LayoutCtx, Storable, packing::PackedSlot};
+
+        Self::load(&PackedSlot(slot_value), U256::ZERO, LayoutCtx::FULL)
+            .expect("unable to decode Pool from slot")
```

Comment:
Same thing here, can we add explicit comments/docs as to why this cant panic? Just makes this more visible/easier to reason when reviewing or interacting with this rather than assuming this is safe or having to check under the hood.